prog/weavewait/main.go

```diff
@@ -39,3 +45,33 @@ func checkErr(err error) {
 		os.Exit(1)
 	}
 }
+
+// Prepend the addresses to /etc/hosts
+func updateHosts(iface *net.Interface) {
+	addrs, err := iface.Addrs()
+	checkErr(err)
+	if len(addrs) == 0 {
+		return
+	}
+	hostname, err := os.Hostname()
+	checkErr(err)
+	var oldHosts, newHosts bytes.Buffer
+	for _, addr := range addrs {
+		fmt.Fprint(&newHosts, strings.Split(addr.String(), "/")[0], "\t", hostname, "\n")
```

Comment:
Might be simpler to type-check addr as an `IPNet` and fetch the address directly; if we encounter some other type of address at this point we don't really want to go string-manipulating it.